flexget/manager.py

```diff
@@ -196,6 +196,11 @@ def initialize(self):
 
         if db_schema.upgrade_required():
             log.info('Database upgrade is required. Attempting now.')
+            if self.options.backup_db:
+                log.verbose('backup db on upgrade flag detected, trying to backup DB')
+                fire_event('manager.backup_db', self)
+            else:
+                log.info('DB is upgrading without a backup, consider using the `--backup-db-on-upgrade` flag')
```

Comment:
Yes, right now you have the backup plugin as a plugin, but have functionality mixed into the core. I think we should separate all functionality related to the backup plugin into the backup plugin.